plugin-server/src/main/ingestion-queues/kafka-queue.ts

```
@@ -180,6 +189,23 @@ export class KafkaQueue {
             .some(({ topic, partitions }) => topic === targetTopic && (!partition || partitions.includes(partition)))
     }
 
+    async isHealthy(): Promise<boolean> {
+        // Uses the heartbeat and group description to decide if the consumer is
+        // in a recoverable state.
+        const sessionTimeout = this.consumerConfig?.sessionTimeout || 30000
+        if (this.lastHeartbeat && Date.now() - this.lastHeartbeat < sessionTimeout) {
+            return true
+        }
+
+        // Consumer has no heartbeat, but maybe it's because the group is currently rebalancing
+        try {
+            const { state } = await this.consumer.describeGroup()
+            return ['CompletingRebalance', 'PreparingRebalance'].includes(state)
+        } catch (err) {
```

What errors are we swallowing here? Why do we want to do so?